### DISCUSSION TITLE
Getting Error While Doing Setup for Elastic Search (Sunbird Learn)

### DISCUSSION THREAD

**Discussion by mahajanmahesh935** [Sunbird-Lern/Community#150] [Installation] [answered]
I am Refering Sunbird Learn Installation document. i followed the steps as it is. but when i go for health check then elasic search's health is showing False.

**Solutions I Tried :**  
1. I made changes in json file of indices creation for elastic Search.
i updated number_of_replica to 0
2. And made changes in max_gram And min_gram Values Because it was showing Differance between max_gram and min_gram must be less than or equal to 1.
now when i am checking cluster health then all the indices have green health but the overall health of elastic search is still false 

{
"id": "api.all.health",
"ver": "health",
"ts": "2023-08-08 11:55:50:482+0530",
"params": {
"resmsgid": "8cb59868-68aa-415c-b9c0-c685f93f425c",
"msgid": "8cb59868-68aa-415c-b9c0-c685f93f425c",
"err": null,
"status": "SUCCESS",
"errmsg": null
},
"responseCode": "OK",
"result": {
"response": {
"checks": [
{
"err": "",
"healthy": true,
"name": "UserOrg service",
"errmsg": ""
},
{
"err": "",
"healthy": true,
"name": "Actor service",
"errmsg": ""
},
{
"err": 500,
"healthy": false,
"name": "Cassandra service",
"errmsg": "Keyspace 'sunbird' does not exist"
},
{
"err": null,
"healthy": false,
"name": "Elastic search service",
"errmsg": null
}
],
"healthy": false,
"name": "Complete health check api"
}
}
}

**Comment by Krishnaj20** [comment]
@pallakartheekreddy - can you check this issue?

**Reply by pallakartheekreddy** [reply]
@AmiableAnil Looks like it is Lern installation issue. Can you check this issue.

**Reply by mahajanmahesh935** [reply]
@pallakartheekreddy @Krishnaj20 Thanks for your response . @AmiableAnil Do you have any input..?

**Comment by BharathwajShankar** [comment]
It appears that you have not set up Cassandra correctly, @mahajanmahesh935. The setup you performed was inadequate for Cassandra, causing it to fail at that point and preventing further progress. We recommend that you meticulously follow the documentation provided in the link below, step by step, without skipping any instructions. This should help you resolve the issue:

https://github.com/Sunbird-Lern/userorg-service/tree/release-5.4.0#readme

**Reply by mahajanmahesh935** [reply]
Thank you @BharathwajShankar . i will try by executing all the steps

**Reply by BharathwajShankar** [reply]
@mahajanmahesh935 were you able to do the health check API successfully? if not can you please let us know what help you require or where you are stuck at?If you are able to get through successfully please mark the thread as answered and close.

**Reply by mahajanmahesh935** [reply]
I have Followed All steps for setup. Health check for cassandra is true but while creating instance of elastic search  the command provided in documentation is giving error so i used a docker compose file to create instance but still health of elastic search is false.

docker-compose file i used : 

version: '3'
services:
  sunbird_es:
    image: docker.elastic.co/elasticsearch/elasticsearch:6.8.11
    container_name: sunbird_es
    ports:
      - "9200:9200"
    volumes:
      - $sunbird_dbs_path/es/data:/usr/share/elasticsearch/data
      - $sunbird_dbs_path/es/logs:/usr/share/elasticsearch/logs
      - $sunbird_dbs_path/es/backups:/opt/elasticsearch/backup
    environment:
      - discovery.type=single-node
    networks:
      - sunbird_db_network
networks:
  sunbird_db_network:
    external: true

i have created the indices and did mapping of it also

**Reply by BharathwajShankar** [reply, marked as answer]
@mahajanmahesh935 can you please check the below environment variables have been set properly?

export sunbird_es_host="localhost"
export sunbird_es_port="9200"
export sunbird_es_cluster="test"

you can set this in bashrc file and retry once.

**Reply by mahajanmahesh935** [reply]
Thank You @BharathwajShankar . error is solved
